### PR TITLE
Nuke disk suicide, better log messages

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
rscadd: Please do not commit suicide with the nuclear authentication disk.
/:cl:
- Nuke disk suicide
- Also, made a difference in the admin logs between a disk that has been
  destroyed, and one that has been taken out of bounds
- I originally wanted to change the nuke time to world.time, but that's not possible without fiddling with the gui, and if I'm changing the GUI I may as well throw it out and put a TGUI in, but I'm not doing that until I get a proper machine

As always currently, need some testing.

### CODE REVIEW

**Comment by TehZombehz** [comment]
I swear once upon a time that the lore was that the station didn't know the disk was even on the station or existed or how it got there.

File: code/game/gamemodes/nuclear/nuclearbomb.dm [hunk 1]
```diff
@@ -439,20 +435,33 @@ This is here to make the tiles around the station mininuke change when it's arme
...
+	var/orig_color = user.color
+	while(world.time < end_time)
+		if(!user)
+			return
+		user.color = RANDOM_COLOUR
+		sleep(1)
+	user.color = orig_color
+	user.visible_message("<span class='suicide'>[user] was destroyed \
+		by the nuclear blast!</span>")
+	return OXYLOSS
```
Comment (lordpidey) [inline code/game/gamemodes/nuclear/nuclearbomb.dm]: Change it to TOXLOSS|BRUTELOSS|FIRELOSS to simulate being exploded/burnt/irradiated

Comment (coiax) [inline code/game/gamemodes/nuclear/nuclearbomb.dm]: Nah, it simulates the actual effects of the nuke going off, which is dying suddenly with no real reason. :3

File: code/game/gamemodes/nuclear/nuclearbomb.dm [hunk 2]
```diff
@@ -462,8 +471,21 @@ This is here to make the tiles around the station mininuke change when it's arme
...
-		message_admins("[src] has been destroyed in ([diskturf ? "[diskturf.x], [diskturf.y] ,[diskturf.z] - <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[diskturf.x];Y=[diskturf.y];Z=[diskturf.z]'>JMP</a>":"nonexistent location"]). Moving it to ([targetturf.x], [targetturf.y], [targetturf.z] - <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[targetturf.x];Y=[targetturf.y];Z=[targetturf.z]'>JMP</a>).")
-		log_game("[src] has been destroyed in ([diskturf ? "[diskturf.x], [diskturf.y] ,[diskturf.z]":"nonexistent location"]). Moving it to ([targetturf.x], [targetturf.y], [targetturf.z]).")
+		return targetturf
 	else
 		throw EXCEPTION("Unable to find a blobstart landmark")
+
+/obj/item/weapon/disk/nuclear/Destroy(force)
+	var/turf/diskturf = get_turf(src)
+
+	if(force)
```
Comment (lzimann) [inline code/game/gamemodes/nuclear/nuclearbomb.dm]: does this mean the disk won't respawn?

Comment (coiax) [inline code/game/gamemodes/nuclear/nuclearbomb.dm]: The force argument was present before this PR during my changes to the GC that allowed for normally immortal items to be force deleted. It shouldn't happen in normal gameplay.

**Pull request merged**